Sources/EmbraceCore/Capture/Network/NetworkPayloadCapture/EncryptedNetworkPayload.swift

```diff
@@ -0,0 +1,202 @@
+
+        case matchedUrl = "matched-url"
+        case sessionId = "session-id"
+
+        case requestBody = "request-body"
+        case requestBodySize = "request-body-size"
+        case requestQuery = "request-query"
+        case requestHeaders = "request-headers"
+
+        case responseBody = "response-body"
+        case responseBodySize = "response-body-size"
+        case responseHeaders = "response-headers"
+        case responseStatus = "response-status"
+
+        case errorMessage = "error-message"
+    }
+
+    init?(
+        request: URLRequest?,
+        response: URLResponse?,
+        data: Data?,
+        error: Error?,
+        startTime: Date?,
+        endTime: Date?,
+        matchedUrl: String,
+        sessionId: String?
+    ) {
+        guard let request = request,
+              let url = request.url,
+              let method = request.httpMethod else {
+            return nil
+        }
+
+        self.url = url.absoluteString
+        self.httpMethod = method
+
+        self.startTime = startTime?.nanosecondsSince1970Truncated
+        self.endTime = endTime?.nanosecondsSince1970Truncated
+
+        self.matchedUrl = matchedUrl
+        self.sessionId = sessionId
+
+        if let body = request.httpBody {
+            self.requestBody = String(data: body, encoding: .utf8)
+            self.requestBodySize = body.count
+        } else {
+            self.requestBody = nil
+            self.requestBodySize = nil
+        }
+
+        self.requestQuery = url.query
+        self.requestHeaders = request.allHTTPHeaderFields ?? [:]
+
+        if let data = data {
+            self.responseBody = String(data: data, encoding: .utf8)
+            self.responseBodySize = data.count
+        } else {
+            self.responseBody = nil
+            self.responseBodySize = nil
+        }
+
+        if let response = response as? HTTPURLResponse {
+            self.responseHeaders = response.allHeaderFields as? [String: String] ?? [:]
+            self.responseStatus = response.statusCode
+        } else {
+            self.responseHeaders = nil
+            self.responseStatus = nil
+        }
+
+        if let error = error {
+            self.errorMessage = error.localizedDescription
+        } else {
+            self.errorMessage = nil
+        }
+    }
+
+    func encrypted(withKey key: String) -> EncryptedPayloadResult? {
+
+        // encode json payload
+        var data: Data?
+        do {
+            data = try JSONEncoder().encode(self)
+        } catch {
+            Embrace.logger.debug("Error encoding `EncryptedNetworkPayload`:\n\(error.localizedDescription)")
+        }
+        guard let data = data else {
+            return nil
+        }
+
+        // generate symmetric key
+        let symmetricKey = SymmetricKey(size: .bits256)
+
+        // encrypt payload using aes
+        var encryptedData: Data?
+        do {
+            encryptedData = try AES.GCM.seal(data, using: symmetricKey).combined
```

Comment:
I think this will read a little better if you move the logic to get the `data` var into this do/catch block for `encryptedData`